ansible/plugins/connection/switch.py

```diff
@@ -54,6 +53,11 @@ def _build_command(self):
 
         self._ssh_command += ['-o', 'GSSAPIAuthentication=no',
                               '-o', 'PubkeyAuthentication=no']
+
+        if 'aos' in self.os_name:
+            self._ssh_command += ['-o', 'KexAlgorithms=+diffie-hellman-group1-sha1',
+                                  '-o', 'HostKeyAlgorithms=+ssh-dss']
```

Comment:
it is strange that aos needs such options, it would be better to explain why these options are needed.